baybe/recommenders/meta/base.py

```diff
@@ -30,19 +49,58 @@ def select_recommender(
     ) -> PureRecommender:
         """Select a pure recommender for the given experimentation context.
 
-        Args:
-            batch_size:
-                See :func:`baybe.recommenders.meta.base.MetaRecommender.recommend`.
-            searchspace:
-                See :func:`baybe.recommenders.meta.base.MetaRecommender.recommend`.
-            objective:
-                See :func:`baybe.recommenders.meta.base.MetaRecommender.recommend`.
-            measurements:
-                See :func:`baybe.recommenders.meta.base.MetaRecommender.recommend`.
-
-        Returns:
-            The selected recommender.
+        See :meth:`baybe.recommenders.base.RecommenderProtocol.recommend` for details
+        on the method arguments.
+        """
+
+    def get_current_recommender(self) -> PureRecommender:
+        """Get the current recommender, if available."""
+        if self._current_recommender is None:
+            raise RuntimeError(
+                f"No recommendation has been requested from the "
+                f"'{self.__class__.__name__}' yet. Because the recommender is a "
+                f"'{MetaRecommender.__name__}', this means no actual recommender has "
+                f"been selected so far. The recommender will be available after the "
+                f"next '{self.recommend.__name__}' call."
+            )
+        return self._current_recommender
+
+    def get_next_recommender(
+        self,
+        batch_size: int,
+        searchspace: SearchSpace,
+        objective: Objective | None = None,
+        measurements: pd.DataFrame | None = None,
+    ) -> PureRecommender:
+        """Get the recommender for the next recommendation.
+
+        Returns the next recommender in row that has not yet been used for generating
+        recommendations. In case of multiple consecutive calls, this means that
+        the same recommender instance is returned until its :meth:`recommend` method
+        is called.
+
+        See :meth:`baybe.recommenders.base.RecommenderProtocol.recommend` for details
+        on the method arguments.
         """
+        # Check if the stored recommender instance can be returned
+        if (
+            self._current_recommender is not None
+            and not self._current_recommender_was_used
+        ):
+            recommender = self._current_recommender
+
+        # Otherwise, fetch the next recommender waiting in row
+        else:
+            recommender = self.select_recommender(
+                batch_size=batch_size,
+                searchspace=searchspace,
+                objective=objective,
+                measurements=measurements,
+            )
+            self._current_recommender = recommender
+            self._current_recommender_was_used = False
```

Comment:
I think @AVHopp has made a good point here. Probably I was rushing too fast given my limited availability. In my thought process, I somehow had in mind that each recommender is gonna be used once, but that is actually not necessarily the case and depends on the particular logic of the meta recommender. For instance, the sequential recommender only skips forward when the data is incremented.

That said, I'm no longer sure if what I did here is correct. Could be, but definitely no guarantee. The fact that the meta recommender tests pass is a good sign, but perhaps they don't catch all necessary aspects. Definitely, they don't cover serialization steps in between.

So I ask you both, @Scienfitz @AVHopp: please give this part another fresh though from scratch and verify what needs to be done. The goal is clear:
* We need the be able to access the recommender for `get_surrogate` before it might have been fit
* When recommending after that step, no recommender in sequence (and don't necessary mean a `SequentialRecommender` here but just in general the next recommender that would have been selected by the meta recommender) should be skipped before it actually produced a recommendation
* We need to be able to de-/serialize at all times during that process and continue where we left

Thanks, guys!

Reply:
it appears to me whether a recommender was used or not needs to be stored in `PureRecommender`, in this manner the object itself knows whether it was used, so `MetaRecommender` can simply ask the current `PureRecommender` whether it was used or not

Reply:
Hey Martin, just saw your comment. I think this would lead to wrong behavior (see my comment to Alex who asked the same) plus has the disadvantage that it makes our pure recommenders stateful. The "was-used" property really needs to be understood on the level of the Meta recommender. It doesn't matter if the recommender object has been used elsewhere or maybe even in parallel.

Reply:
then I think where the recommender was sued flag is reset cannot be handled by the base class and instead prob needs to go into the respective meta recommender implementation
seems to me only there is can be decided whether a new recommender object is returned or the old oneor du you want to do some hash-based check in the mother class?

Reply:
ok so next try, pushed a commit that

Gives the meta recommender a private set of integers, which remembers the `id`s of recommenders that were used. Ids are only added after successful `.reommend` call. This
- does not modify pure recommenders so they remain stateless
- should seamlessly deal with
  - situations where recommender was selected but not actually used
  - situations where a recommender might have been used previously in the sequence, eg via a cyclic iterator 
- required a small change to the cattrs converter, as by default it doesnt unstructure `set`s

Reply:
Looks good to me. Have officially re-approved, feel free to resolve here.